### PR TITLE
photon: 1.0.7 -> 1.3.0

### PR DESCRIPTION
**Pull request by bignaux**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bignaux** [comment]
@GrahamcOfBorg build photon

File: pkgs/development/python-modules/tld/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+buildPythonPackage rec {
+  pname   = "tld";
+  version = "0.9.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0i0prgwrmm157h6fa5bx9wm0m70qq2nhzp743374a94p9s766rpp";
+  };
+
+  doCheck = false;
```
Comment (flokli) [inline pkgs/development/python-modules/tld/default.nix]: can we run the tests?

File: pkgs/tools/networking/photon/default.nix [hunk 2]
```diff
@@ -1,32 +1,27 @@
-{ stdenv, pythonPackages, fetchurl, makeWrapper }:
+{ stdenv, python3Packages, fetchFromGitHub, makeWrapper }:
 
-with pythonPackages;
-buildPythonApplication rec {
-  pname = "photon";
-  version = "1.0.7";
+python3Packages.buildPythonPackage rec {
```
Comment (flokli) [inline pkgs/tools/networking/photon/default.nix]: Is photon now meant to be used as a library?

If not, please keep using `buildPythonApplication`.

File: pkgs/tools/networking/photon/default.nix [hunk 3]
```diff
@@ -1,32 +1,27 @@
...
 
-  src = fetchurl {
-    url = "https://github.com/s0md3v/Photon/archive/v${version}.tar.gz";
-    sha256 = "0c5l1sbkkagfxmh8v7yvi6z58mhqbwjyr7fczb5qwxm7la42ah9y";
-  };
+  version = "1.3.0";
+  name = "photon-${version}";
```
Comment (flokli) [inline pkgs/tools/networking/photon/default.nix]:
```suggestion
  pname = "photon";
```

**Review by flokli — CHANGES_REQUESTED** [review]
please make that two commits:

 - pythonPackages.tld: init at 0.9.3
 - pythonPackages.photon: 1.0.7 -> 1.3.0

On the other hand, it might make sense to move `photon` to `all-packages.nix`, so `photon: 1.0.7 -> 1.3.0` (as it seems to be just an application)

File: pkgs/tools/networking/photon/default.nix (hunk 2 above)
Comment (bignaux) [reply pkgs/tools/networking/photon/default.nix]: no but i need python3.

File: pkgs/tools/networking/photon/default.nix (hunk 3 above)
Comment (bignaux) [reply pkgs/tools/networking/photon/default.nix]: agree :)

File: pkgs/development/python-modules/tld/default.nix (hunk 1 above)
Comment (bignaux) [reply pkgs/development/python-modules/tld/default.nix]: no, doesn't work. I asked on irc, one commit for that would be ok.

File: pkgs/tools/networking/photon/default.nix (hunk 2 above)
Comment (flokli) [reply pkgs/tools/networking/photon/default.nix]: And why can't you use `pythonPackages.buildPythonApplication`?

Comment (bignaux) [reply pkgs/tools/networking/photon/default.nix]: i did finally. was a mistake.

File: pkgs/development/python-modules/tld/default.nix (hunk 1 above)
Comment (flokli) [reply pkgs/development/python-modules/tld/default.nix]: You mean, the tld tests are broken upstream as well? If yes, please open an issue there and add a comment with a link to it next to this line.

Comment (bignaux) [reply pkgs/development/python-modules/tld/default.nix]: No they are required others python modules and i don't even use photon anymore ...

Comment (flokli) [reply pkgs/development/python-modules/tld/default.nix]: So these modules would be a candidate for `checkInputs`.

Running unit tests for interpreted languages like python are important to catch breakages when dependencies are bumped, even more than with compiled languages.

Comment (bignaux) [reply pkgs/development/python-modules/tld/default.nix]: https://github.com/barseghyanartur/tld/tree/master/requirements too much works i look earlier. that failed on pytest-cov

**Comment by flokli** [comment]
@GrahamcOfBorg build python.pkgs.tld python3.pkgs.tld

File: pkgs/development/python-modules/tld/default.nix (hunk 1 above)
Comment (flokli) [reply pkgs/development/python-modules/tld/default.nix]: I enabled the tests, had to add the required testing libraries to `checkInputs`, and exclude some tests that seem to try to use networking.

**Comment by risicle** [comment]
Builds and appears to work for me, non-nixos linux x86_64

**Comment by veprbl** [comment]
@GrahamcOfBorg build python.pkgs.tld python3.pkgs.tld photon

**Review by veprbl — APPROVED** [review]

**Pull request merged**